nbsite/pyodide/__init__.py

```diff
@@ -177,6 +179,40 @@ def _model_json(model: Model, target: str) -> Tuple[Document, str]:
         version   = BOKEH_VERSION,
     ))
 
+extra_resources = defaultdict(lambda: {'js': [], 'css': [], 'js_modules': {}, 'js_exports': {}})
+
+def write_resources(out_dir, source, resources):
+    if sys.platform == "win32":
+        extra_resources[source]['css'] += resources['css']
+        extra_resources[source]['js'] += resources['js']
+        extra_resources[source]['js_exports'].update(resources['js_exports'])
+        extra_resources[source]['js_modules'].update(resources['js_modules'])
+        return
+
+    import fcntl
+    out_path = pathlib.Path(str(out_dir))
+    out_path.mkdir(exist_ok=True)
+    resources_file = out_path / RESOURCE_FILE
+    existing = resources_file.is_file()
+    with open(resources_file, 'a+', encoding='utf-8') as rfile:
+        fcntl.flock(rfile, fcntl.LOCK_EX)
+        rfile.seek(0)
+        if existing:
+            all_resources = json.load(rfile)
+            rfile.seek(0)
+            rfile.truncate()
+        else:
+            all_resources = {}
+        if source in all_resources:
+            source_resources = all_resources[source]
+            source_resources['css'] += [css for css in resources['css'] if css not in source_resources['css']]
+            source_resources['js'] += [js for js in resources['js'] if js not in source_resources['js']]
+            source_resources['js_exports'].update(resources['js_exports'])
+            source_resources['js_modules'].update(resources['js_modules'])
+        else:
+            all_resources[source] = source_resources = resources
```

Comment:
I don't follow the `source_resource` assignment here.

Reply:
Will leave some comments.